cnf-tests/pod-utils/hugepages-allocator/main.go

```diff
@@ -0,0 +1,72 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"math"
+	"os"
+	"syscall"
+	"time"
+	"unsafe"
+
+	"k8s.io/klog/v2"
+)
+
+const (
+	// MapHugeShift Shift for HugePage size
+	MapHugeShift = 26
+	// DefaultHugePageSize 1GB HugePage size
+	DefaultHugePageSize = 1 * 1024 * 1024 * 1024
+)
+
+// MAP_HUGE_1GB 1GB HugePage mmap flag
+const MAP_HUGE_1GB = 30 << MapHugeShift
+
+type Args struct {
+	Sleep        time.Duration
+	HugePageSize int
+}
+
+func main() {
+	klog.InitFlags(nil)
+	args := &Args{}
+	flag.DurationVar(&args.Sleep, "timeout", math.MaxInt64, "set the sleep in seconds for program to sleep - sleep forever by default")
+	flag.IntVar(&args.HugePageSize, "hugepage-size", DefaultHugePageSize, "hugepage size to allocate - allocate 1G by default")
+
+	flag.Parse()
+
+	// Flags for HugePage allocation
+	mmapFlags := syscall.MAP_PRIVATE | syscall.MAP_ANONYMOUS | syscall.MAP_HUGETLB | MAP_HUGE_1GB
+	// Use mmap to allocate memory
+	addr, _, errno := syscall.Syscall6(
+		syscall.SYS_MMAP,
+		0,                          // Let the kernel choose the address
+		uintptr(args.HugePageSize), // Size of the memory
+		uintptr(syscall.PROT_READ|syscall.PROT_WRITE), // Read/Write permissions
+		uintptr(mmapFlags), // mmap flags
+		0,                  // File descriptor (not used for anonymous memory)
+		0,                  // Offset
+	)
+	if errno != 0 {
+		// there's little we can do here
+		_, _ = fmt.Fprintf(os.Stderr, "Failed to allocate hugepage. errno=%v\n", errno)
+		os.Exit(1)
+	}
+	memory := unsafe.Pointer(addr)
+	// Write a byte to the allocated memory
+	*(*byte)(memory) = 42
+
+	klog.InfoS("Successfully allocated 1GB HugePage memory", "address", fmt.Sprintf("%p", unsafe.Pointer(addr)))
+
+	klog.InfoS("Sleeping", "period", args.Sleep.String())
+	time.Sleep(args.Sleep)
```

Comment:
to make this one right we need to have a sig term channel and when we remove the pod it gets called and we do the cleanup before we exist this app

Reply:
From where we should get the signal?

Reply:
https://github.com/GoogleCloudPlatform/golang-samples/blob/main/run/sigterm-handler/main.go#L48

some thing like that so after the test we delete that pod it will have the opportunity to clean the RAM

Reply:
When I deleted the pod the memory got cleared, but we can add it anyway, just in case